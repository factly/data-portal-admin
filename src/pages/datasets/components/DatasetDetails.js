import React from 'react';
import { useHistory } from 'react-router-dom';
import { Descriptions, Card } from 'antd';
import moment from 'moment';
import { EditOutlined } from '@ant-design/icons';

const DatasetDetails = ({ dataset }) => {
  const history = useHistory();
  return (
    <Card>
      <Descriptions
        title={dataset.title}
        column={1}
        extra={
          <EditOutlined
            key="edit"
            style={{ fontSize: '150%' }}
            onClick={() => history.push(`/datasets/${dataset.id}/edit`)}
          />
        }
        bordered
      >
        <Descriptions.Item label="Organisation">{dataset.organisation}</Descriptions.Item>
        <Descriptions.Item label="Sectors">{dataset.sectors}</Descriptions.Item>
        <Descriptions.Item label="Contact email">{dataset.contact_email}</Descriptions.Item>
        <Descriptions.Item label="Contact name">{dataset.contact_name}</Descriptions.Item>
        <Descriptions.Item label="Data standard">{dataset.data_standard}</Descriptions.Item>
        <Descriptions.Item label="Description">
          <div
            style={{ whiteSpace: 'pre-wrap' }}
            dangerouslySetInnerHTML={{ __html: dataset.description }}
          />
        </Descriptions.Item>
        <Descriptions.Item label="Units">
          <div
            style={{ whiteSpace: 'pre-wrap' }}
            dangerouslySetInnerHTML={{ __html: dataset.units }}
          />
        </Descriptions.Item>
        <Descriptions.Item label="Frequency">{dataset.frequency}</Descriptions.Item>
        <Descriptions.Item label="Granularity">{dataset.granularity}</Descriptions.Item>
        <Descriptions.Item label="License">{dataset.license}</Descriptions.Item>
        <Descriptions.Item label="Related articles">{dataset.related_articles}</Descriptions.Item>
        <Descriptions.Item label="Source">{dataset.source}</Descriptions.Item>
        <Descriptions.Item label="Source Link">{dataset.source_link}</Descriptions.Item>
        <Descriptions.Item label="Archive Link">{dataset.archive_link}</Descriptions.Item>
        <Descriptions.Item label="Temporal coverage">{dataset.temporal_coverage}</Descriptions.Item>
        <Descriptions.Item label="Time saved">{dataset.time_saved}</Descriptions.Item>
        <Descriptions.Item label="Profiling URL">
          <a>{dataset.profiling_url}</a>
        </Descriptions.Item>
        <Descriptions.Item label="Is Public">{dataset.is_public ? 'yes' : 'no'}</Descriptions.Item>
        <Descriptions.Item label="Created At">
          {moment(dataset.created_at).format('Do MMMM YYYY')}
        </Descriptions.Item>
        <Descriptions.Item label="Last Updated At">
          {moment(dataset.updated_at).format('Do MMMM YYYY')}
        </Descriptions.Item>
        <Descriptions.Item label="Next Update">
          {dataset.next_update ? moment(dataset.next_update).format('Do MMMM YYYY') : null}
        </Descriptions.Item>
      </Descriptions>
    </Card>
  );
};

export default DatasetDetails;
