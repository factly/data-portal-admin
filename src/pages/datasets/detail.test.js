import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { shallow } from 'enzyme';

import DatasetDetails from './detail';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}));

describe('Datasets detail component', () => {
  beforeEach(() => {
    useParams.mockReturnValueOnce({ id: 1 });
  });
  it('should render the component', () => {
    useSelector.mockReturnValueOnce({ id: 1, name: 'dataset' });
    const component = shallow(<DatasetDetails />);
    expect(component).toMatchSnapshot();
  });
  it('should render the component without dataset', () => {
    useSelector.mockReturnValueOnce({});
    const component = shallow(<DatasetDetails />);
    expect(component).toMatchSnapshot();
  });
});
