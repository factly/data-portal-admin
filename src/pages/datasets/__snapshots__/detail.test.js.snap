// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datasets detail component should render the component 1`] = `
<Fragment>
  <DatasetDetails
    dataset={
      Object {
        "id": 1,
        "name": "dataset",
      }
    }
  />
  <DatasetFormatsList
    datasetId={1}
    showOperations={false}
  />
</Fragment>
`;

exports[`Datasets detail component should render the component without dataset 1`] = `
<Fragment>
  <DatasetDetails
    dataset={Object {}}
  />
  <DatasetFormatsList
    showOperations={false}
  />
</Fragment>
`;
